src/Metric/MetricFactory.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Metric;
+
+use Baldinof\RoadRunnerBundle\Exception\BadConfigurationException;
+use Baldinof\RoadRunnerBundle\Exception\UnknownRpcTransportException;
+use Spiral\Goridge\RPC;
+use Spiral\Goridge\SocketRelay;
+use Spiral\RoadRunner\Metrics;
+use Spiral\RoadRunner\MetricsInterface;
+
+final class MetricFactory
+{
+    /**
+     * @var bool
+     */
+    private $metricsEnabled;
+
+    /**
+     * @var string
+     */
+    private $rrRpc;
+
+    /**
+     * @var string
+     */
+    private $kernelProjectDir;
+
+    public function __construct(string $rrRpc, string $kernelProjectDir, bool $metricsEnabled)
+    {
+        $this->rrRpc = $rrRpc;
+        $this->kernelProjectDir = $kernelProjectDir;
+        $this->metricsEnabled = $metricsEnabled;
+    }
+
+    public function getMetricService(): MetricsInterface
+    {
+        $nullMetrics = new NullMetrics();
+
+        if (!$this->metricsEnabled) {
+            return $nullMetrics;
+        }
```

Comment:
This could be inlined.

```suggestion
        if (!$this->metricsEnabled) {
            return new NullMetrics();
        }
```